### DISCUSSION TITLE
Keep input field values when following link and going back

### DISCUSSION THREAD

**Discussion by LaundroMat** [unpoly/unpoly#649] [Q&A]
I have an item-list page, where users can select items in the list (with a checkbox, to e.g. delete multiple items at once). On clicking an item, an item-detail page (with a form to change the item) is opened, meaning that if they go back to the list view, their selection is reset.

I tried solving this by opening the item-detail page in a new layer. That way, if the user dismisses the page , their selection on the root item-list page is not reset.

The problem then is that the items in item-list are not updated if a user has changed something in item-detail. I could catch the dismiss event, but that'd mean a page reload, unchecking the selected items.

There's also the fact that the browser back button ([for good reasons](https://github.com/unpoly/unpoly/discussions/434)) does not dismiss the overlay but reloads the previous page in history (users can go back and forth between items in the items-detail page too...).

What would be a good strategy to implement keeping the state of checkboxes on the item-list page when opening and closing/going back from the item-detail page?

**Comment by hpvd** [comment]
which version of unpoly are you using?
There was recently in v3.8 some nice work on browser back button/history see https://unpoly.com/changes/3.8.0#improvements-to-history-restoration
not sure if this is related, but I have opened an issue in this field... https://github.com/unpoly/unpoly/issues/647

**Comment by adam12** [comment]
I suspect you'll need to use some other mechanism to preserve the state. This could be something inside `localStorage` or on a server somewhere.

**Comment by triskweline** [comment]
How about storing the selection state in the URL of the list layer? This way selection would be preserved when the list is reloaded, or when the URL is restored after history navigation.

Updating the URL on the client could work like this (untested):

```js
up.on('change', 'form#item-list', function(event, form) {
  let params = up.Params.fromForm(form)
  let urlWithParams = location.path + '?' + params.toQuery()
  up.history.replace(urlWithParams)
})
```
